src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -658,15 +658,18 @@ public static double MaxMagnitude(double x, double y)
             }
 
             // We do this comparison first and separately to handle the -0.0 to +0.0 comparision
-            // * Doing (x < y) first could get transformed into (y >= x) by the JIT which would
+            // * Doing (ax < ay) first could get transformed into (ay >= ax) by the JIT which would
             //   then return an incorrect value
 
-            if (x == y)
+            double ax = Abs(x);
+            double ay = Abs(y);
+
+            if (ax == ay)
             {
                 return double.IsNegative(x) ? y : x;
             }
 
-            return (Abs(x) < Abs(y)) ? y : x;
+            return (ax < ay) ? y : x;
```

Comment:
Looking at the spec in https://github.com/dotnet/corefx/issues/31903

```
public static double MinMagnitude(double, double);             // IEEE `maxNumMag` - Equivalent to `Math.Max(Math.Abs(x), Math.Abs(y))`
```

Shouldn't this be returning `ax` or `ay`?

Reply:
Same response as https://github.com/dotnet/corefx/pull/33359#discussion_r233495239, the speclet is incorrect and doesn't correctly indicate that it returns the original (non absolute) value of x or y.